src/docfx/build/xref/XrefResolver.cs

```diff
@@ -215,12 +212,38 @@ orderby monikerGroup
         return model;
     }
 
-    private IReadOnlyDictionary<string, InternalXrefSpec[]> BuildInternalXrefMap()
+    public Dictionary<string, ExternalXrefSpec> ResolveXrefMapByFile(
```

Comment:
Presumably we don't need this new method to mimic what SDP is doing [here](https://github.com/dotnet/docfx/blob/v3/src/docfx/lib/schema/JsonSchemaTransformer.cs#L466) to produce an xrefmap. 

For conceptual, we only need the definitions so call `ResolveXrefSpec` if `uid` is specified in YAML header.

Reply:
@yufeih  
For current implementation, the xrefs per file are totally independent from SDP `_xrefmap`. 
And the xrefs-per-file are generated in the last stage instead of during the file processing, it is easy and clean.